### PR TITLE
refactor(js,client): remove superstruct

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Remove superstruct, replaced with own type check functions.

The package "superstruct" is "es module" and can not be "required" in some circumstances, we can fix it by setting "alias", but our users will have that problem too.

## Testing

UTs.

### CODE REVIEW

File: packages/client/src/types/index.ts [hunk 1]
```diff
@@ -11,20 +10,46 @@ export type LogtoConfig = {
...
+      typeof value.token !== 'string' ||
+      typeof value.scope !== 'string' ||
+      typeof value.expiresAt !== 'number'
+    ) {
+      return false;
+    }
+  }
 
-export type LogtoSignInSessionItem = Infer<typeof LogtoSignInSessionItemSchema>;
+  return true;
```
Comment (charIeszhao) [inline packages/client/src/types/index.ts:47]: Direct return the result?
```
return typeof value.token !== 'string' || typeof value.scope !== 'string' || typeof value.expiresAt !== 'number';
```

File: packages/client/src/types/index.ts [hunk 2]
```diff
@@ -11,20 +10,46 @@ export type LogtoConfig = {
...
 
-export const LogtoSignInSessionItemSchema = type({
-  redirectUri: string(),
-  codeVerifier: string(),
-  state: string(),
-});
+  for (const value of Object.values(data)) {
+    if (!isArbitraryObject(value)) {
+      return false;
+    }
```
Comment (charIeszhao) [inline packages/client/src/types/index.ts:36]: What if the map is empty?

File: packages/client/src/types/index.ts (hunk 1 above)
Comment (wangsijie) [reply packages/client/src/types/index.ts]: updated with array.every

File: packages/client/src/types/index.ts (hunk 2 above)
Comment (wangsijie) [reply packages/client/src/types/index.ts]: Empty map is valid?

**Review by gao-sun — APPROVED** [review]

**Review by charIeszhao — APPROVED** [review]

File: packages/client/src/types/index.ts (hunk 2 above)
Comment (charIeszhao) [reply packages/client/src/types/index.ts]: Hmmm, you're right. No problem in here

**Pull request merged**